api/tasks.py

```diff
@@ -11,3 +14,22 @@ def create_delivery(order):
     else:
         delivery.print_pdf()
         return {'success': 'all good'}
+
+
+@app.task
+def update_customer_if_is_partner(email):
+    if _check_customer_is_partner(email):
+        # here will come the code to update the Wordpress user role.
```

Comment:
IMO this should be done in a separate Celery task. This is the exact best-practice Heroku's docs describe in https://devcenter.heroku.com/articles/celery-heroku#small-short-lived-tasks. We should keep this in mind.

Reply:
Good practice yeah keeping them simple tasks.